react/src/pages/ChatPage.tsx

```diff
@@ -41,32 +63,239 @@ function useDefaultEndpointId() {
   return endpoint_list?.items[0]?.endpoint_id || undefined;
 }
 
-const ChatPage: React.FC<ChatPageProps> = () => {
-  const [{ endpointId, modelId, agentId }] = useQueryParams({
+function useChatProviderData(defaultEndpointId?: string): ChatProviderData {
+  const [{ endpointId, modelId, agentId, apiKey }] = useQueryParams({
     endpointId: StringParam,
     agentId: StringParam,
     modelId: StringParam,
+    apiKey: StringParam,
   });
 
-  const defaultEndpointId = useDefaultEndpointId();
-
-  const conversation: ConversationType = {
-    id: useId(),
-    label: t('webui.menu.Chat'),
-    chats: [],
-  };
-
-  const provider: ChatProviderType = {
-    basePath: 'v1',
-    agentId: agentId ?? undefined,
+  return {
+    basePath: 'v1', // Use OpenAPI 'v1' for OpenAI compatibility basePath,
+    baseURL: '',
     endpointId: endpointId ?? defaultEndpointId ?? undefined,
+    agentId: agentId ?? undefined,
     modelId: modelId ?? undefined,
+    apiKey: apiKey ?? undefined,
   };
+}
+
+interface ChatHistoryDrawerProps {
+  history: ChatHistoryData[];
+  open?: boolean;
+  onClickClose: () => void;
+  onClickRemove: (id: string) => void;
+  onClickHistory: (id: string) => void;
+}
+
+const ChatHistoryDrawer = ({
+  history,
+  open,
+  onClickClose,
+  onClickRemove,
+  onClickHistory,
+}: ChatHistoryDrawerProps) => {
+  return (
+    <Drawer
+      getContainer={false}
+      open={open}
+      onClose={onClickClose}
+      mask={false}
+      maskClosable={true}
+      title={t('chatui.History')}
+    >
```

Comment:
Trash button looks too far off the right.
```suggestion
     styles={{
        body: { paddingRight: 0 },
      }}
    >
```
Before:
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/1a844bbb-04c7-4b96-b093-cc0fc1bc3e3d.png)

After:
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/edbb9f07-4c1c-48f0-9d90-852c6450c303.png)